browser-sign-in/src/androidTest/java/com/okta/android/samples/browser_sign_in/test/TestClientBuilder.kt

```diff
@@ -0,0 +1,101 @@
+import java.util.Arrays
+import java.util.concurrent.TimeUnit
+import kotlin.String
+
+internal class TestClientBuilder(
+    private val clientId: String,
+    private val orgUrl: String,
+    private val clientCredentials: ClientCredentials<*>
+) {
+    private val clientConfiguration = DefaultClientBuilder().clientConfiguration.apply {
+        baseUrl = orgUrl
+        clientId = this@TestClientBuilder.clientId
+    }
+
+    fun build(): ApiClient {
+        val cacheManager = Caches.newCacheManager()
+            .withDefaultTimeToIdle(clientConfiguration.cacheManagerTti, TimeUnit.SECONDS)
+            .withDefaultTimeToLive(clientConfiguration.cacheManagerTtl, TimeUnit.SECONDS)
+            .build()
+
+        clientConfiguration.baseUrlResolver = DefaultBaseUrlResolver(orgUrl)
+        val apiClient = ApiClient(
+            restTemplate(),
+            cacheManager,
+            clientConfiguration
+        )
+        apiClient.basePath = clientConfiguration.baseUrl
+        clientConfiguration.clientCredentialsResolver = DefaultClientCredentialsResolver(clientCredentials)
+        apiClient.setApiKeyPrefix("SSWS")
+        apiClient.setApiKey(clientCredentials.credentials as String)
+        return apiClient
+    }
+
+    private fun restTemplate(): RestTemplate {
+        val objectMapper = ObjectMapper()
+        objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL)
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+        objectMapper.configure(DeserializationFeature.READ_UNKNOWN_ENUM_VALUES_AS_NULL, true)
+        objectMapper.registerModule(JavaTimeModule())
+        objectMapper.registerModule(JsonNullableModule())
+        val module = SimpleModule()
+        module.addSerializer(UserProfile::class.java, UserProfileSerializer())
+        module.addDeserializer(UserProfile::class.java, UserProfileDeserializer())
+        objectMapper.registerModule(module)
+        val mappingJackson2HttpMessageConverter = MappingJackson2HttpMessageConverter(objectMapper)
+        mappingJackson2HttpMessageConverter.supportedMediaTypes = Arrays.asList(
+            MediaType.APPLICATION_JSON,
+            MediaType.parseMediaType("application/x-pem-file"),
+            MediaType.parseMediaType("application/x-x509-ca-cert"),
+            MediaType.parseMediaType("application/pkix-cert")
+        )
+        val messageConverters: MutableList<HttpMessageConverter<*>> = ArrayList()
+        messageConverters.add(mappingJackson2HttpMessageConverter)
+        val restTemplate = RestTemplate(messageConverters)
+        restTemplate.errorHandler = ErrorHandler()
+        restTemplate.requestFactory = OkHttp3ClientHttpRequestFactory()
+        val uriTemplateHandler = DefaultUriBuilderFactory()
+        uriTemplateHandler.encodingMode = DefaultUriBuilderFactory.EncodingMode.VALUES_ONLY
+        restTemplate.uriTemplateHandler = uriTemplateHandler
+        return restTemplate
+    }
```

Comment:
This is identical logic to https://github.com/okta/okta-sdk-java/blob/a61a2530ddab4dcab4b66cfd338d04ed59eb8089/impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java#L415

The only difference is that I'm setting a different restTemplate.requestFactory here.

Reply:
Is it possible for us to reuse this functionality from the management SDK? It feels like a shame to copy/paste it, since we're pulling it in as a dependency anyway.